README.md

```diff
@@ -207,23 +220,293 @@ More information can be found [here](http://www.diveintopython3.net/strings.html
 | pyVersion        | Returns the version of Python                      | `pyVersion()`                                                        |
 
 
-# Usage Examples   
+# Wrapping Python packages
+
+The following tools help generate R functions that wrap Python functions along with reference documentation (.Rd files) for the wrapper functions. The R documentation is generated from the Sphinx-based Python docstrings.
+
+Prequisites:
+* The Python package is downloaded and installed on the machine
+* The Python package is available on Python search path
+
+A Python package may have multiple modules, each with its own namespace. Each module has its own functions, classes, and variables. Each function or class has its own local namespace. In R, all functions and classes within the same package share a package namespace. To avoid the namespace collisions and allow customization to the wrapping package, we provide the following functions to help you pick which modules, functions, and classes to expose in R:
+
+* `generateRWrappers`
+* `generateRdFiles`
+
+The .Rd file generation must happen at the time the package is built, not at the time the package is loaded. The R function and constructor wrappers must happen at the time the package is loaded. These functions should be called with the similar parameters to ensure that all R wrapped functions have corresponding reference documentation.
+
+## Examples:
+
+In this example, I will demonstrate how we generate [synapser](https://github.com/Sage-Bionetworks/synapser) and [synapserutils](https://github.com/Sage-Bionetworks/synapserutils) packages by wrapping Python package [synapsePythonClient](https://github.com/Sage-Bionetworks/synapsePythonClient).
```

Comment:
how we generate *the* synapser package and *the* synapserutils package by wrapping *the* Python package synapsePythonClient.